tracer/src/Datadog.Trace/Tracer.cs

```diff
@@ -322,6 +325,57 @@ void IDatadogTracer.Write(ArraySegment<Span> trace)
             }
         }
 
+        /// <summary>
+        /// Sets the details of the user on the local root span
+        /// </summary>
+        /// <param name="email">The user's email</param>
+        /// <param name="name">The user's name</param>
+        /// <param name="id">The user's id</param>
+        /// <param name="sessionId">The user's sessionId</param>
+        /// <param name="role">The user's role</param>
+        public void SetUser(
+            string email = null,
+            string name = null,
+            string id = null,
+            string sessionId = null,
+            string role = null)
+        {
+            var scope = Tracer.Instance.ActiveScope;
+
+            if (ScopeManager.Active == null || ScopeManager.Active.Span is not Span span)
+            {
+                Log.Warning("Calling SetUser without an span present");
+                return;
+            }
+
+            var localRootSpan = span.Context.TraceContext?.RootSpan ?? span;
```

Comment:
IMO, _this_ is the API that should be exposed. There are many more applications than just setting the user

Reply:
> Would you see instead this API `SetUser` on `ISpan`

If we go this route, we should consider adding it as an extension method on `ISpan` and not modifying the interface itself. This is not an intrinsic feature of spans (it builds on top of the `SetTag()` API), and adding members to a public interface is a breaking change.

Reply:
Same comment about using an extension method applies to `ITracer`, although adding it at that level seems wrong. It feels like you are adding it to all traces generated by that tracer, not just the current trace.

BTW, there are other tags that are also added to the root span only. We have special handling for those inside `SetTag()`.